dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/Power.mo

```diff
@@ -0,0 +1,48 @@
+within Dynawo.NonElectrical.Blocks.Continuous;
+
+/*
+* Copyright (c) 2021, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block Power "Outputs a power of the input"
+
+  import Modelica;
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Real N "Exponent of the power function";
+
+  final parameter Boolean isNInteger = (N == floor(N));
+
+equation
+  if isNInteger then
```

Comment:
I think you could simplify this with:
y = if isNInteger then u ^ N then abs(u) ^ N;

Reply:
You prefer a single line :
y = if isNInteger then u ^ N else abs(u) ^ N;
?

Reply:
I find the current layout more readable : one line per outcome.